Core/Contributions/Solutions Software/SSW Widget Enhancements.pax

```diff
@@ -106,6 +122,15 @@ TextEdit subclass: #FormattedTextEdit
 
 !Canvas methodsFor!
 
+clipTo: aRectangle during: aBlock
+
+	| oldClipRegion |
+
+	oldClipRegion := Region empty.
+	GDILibrary default getClipRgn: self asParameter hrgn: oldClipRegion asParameter.
```

Comment:
I had to think hard to convince myself that this doesn't leak a region, but I think it is OK. You might want to explicitly free oldClipRegion at the end of your ensure block, however, in order to avoid generating finalizable garbage.